### PR TITLE
feat: Fir 38128 ecosystem support for struct type for go sdk

### PR DESCRIPTION
**Pull request by stepansergeevitch**
- Added support for struct type
- Minor testing improvements

### CODE REVIEW

File: rows.go [hunk 1]
```diff
@@ -126,6 +126,69 @@ func checkTypeValue(columnType string, val interface{}) error {
...
+	if !ok {
+		return nil, fmt.Errorf("unexpected value for struct type: %v", val)
+	}
+	res := make(map[string]driver.Value)
+	if len(fields) != len(structValue) {
+		return nil, fmt.Errorf("expected %d fields, but got %d", len(fields), len(structValue))
+	}
+	for fieldName, fieldType := range fields {
+		if fieldValue, ok := structValue[fieldName]; ok {
+			res[fieldName], _ = parseValue(fieldType, fieldValue)
```
Comment (ptiurin) [inline rows.go:184]: Let's add a comment that we're ignoring error on purpose here

File: rows_test.go
Comment (ptiurin) [inline rows_test.go:1]: Do we check somewhere the type of the column returned (i.e. the metadata)? Is that even a thing in golang?

**Review by ptiurin — APPROVED** [review]
Otherwise lgtm

File: rows_test.go
Comment (stepansergeevitch) [reply rows_test.go:1]: No Go SDK doesn't report metadata, only column names

File: rows.go (hunk 1 above)
Comment (stepansergeevitch) [reply rows.go:184]: Good that you've notices since we actually shouldn't ignore it here :)

**Pull request merged**